sdk/appcenter/src/main/java/com/microsoft/appcenter/persistence/DatabasePersistence.java

```diff
@@ -228,7 +306,19 @@ public String getLogs(@NonNull String group, @IntRange(from = 0) int limit, @Non
                 try {
 
                     /* Deserialize JSON to Log. */
-                    candidates.put(dbIdentifier, getLogSerializer().deserializeLog(values.getAsString(COLUMN_LOG)));
+                    String logPayload;
+                    String databasePayload = values.getAsString(COLUMN_LOG);
+                    if (databasePayload == null) {
+                        File file = getLargePayloadFile(largePayloadGroupDirectory, dbIdentifier);
+                        AppCenterLog.debug(LOG_TAG, "Read payload file " + file);
+                        logPayload = StorageHelper.InternalStorage.read(file);
+                    } else {
+                        logPayload = databasePayload;
+                    }
+                    if (logPayload == null) {
```

Comment:
Move this if statement in above if statement (after line 314). We don't need to even check this if payload is not big.

Reply:
This is to make sure that if `StorageHelper.InternalStorage.read(file);` has an error, it will be managed. `StorageHelper.InternalStorage.read(file);` does not throw exceptions on error, it returns null.

Reply:
I meant if `databasePayload != null` then you don't need this if statement.

Reply:
ok